machine_translation/sampling.py

```diff
@@ -117,7 +117,7 @@ class BleuValidator(SimpleExtension, SamplingBase):
     """Implements early stopping based on BLEU score."""
 
     def __init__(self, source_sentence, samples, model, data_stream,
-                 config, n_best=1, track_n_models=1, trg_ivocab=None,
+                 config, n_best=1, track_n_models=1,
```

Comment:
I can't find a better way to obtain `trg_eos_idx`. We must retrieve `trg_eos_sym` from 
```
sources = self._get_attr_rec(self.main_loop, 'data_stream')
trg_eos_sym = sources.data_streams[1].dataset.eos_token
```

so it's better to obtain `trg_ivocab` through `sources.data_streams[1].dataset.dictionary`. And in the example, there is no function call that pass the `trg_ivocab`.